File: greeting_card.py
#!/usr/bin/env python3
"""
Hello World Greeting Card Generator
A simple Python script that creates and displays greeting card messages
"""

import sys
import time
from datetime import datetime

class GreetingCard:
    """A class to create and display greeting card messages"""
    
    def __init__(self):
        self.border_char = "═"
        self.corner_chars = ["╔", "╗", "╚", "╝"]
        self.side_char = "║"
        
    def create_border(self, width):
        """Create decorative border for the card"""
        top = self.corner_chars[0] + self.border_char * (width - 2) + self.corner_chars[1]
        bottom = self.corner_chars[2] + self.border_char * (width - 2) + self.corner_chars[3]
        return top, bottom
    
    def center_text(self, text, width):
        """Center text within given width"""
        padding = (width - len(text) - 2) // 2
        return f"{self.side_char}{' ' * padding}{text}{' ' * (width - len(text) - padding - 2)}{self.side_char}"
    
    def display_hello_world_card(self):
        """Display the main Hello World greeting card"""
        width = 50
        top_border, bottom_border = self.create_border(width)
        
        print("\n" + "🌟" * 20)
        print(top_border)
        print(self.center_text("", width))
        print(self.center_text("✨ HELLO WORLD GREETING CARD ✨", width))
        print(self.center_text("", width))
        print(self.center_text("Hello World!", width))
        print(self.center_text("", width))
        print(self.center_text("Welcome to this beautiful greeting!", width))
        print(self.center_text("Created with Python and love ❤️", width))
        print(self.center_text("", width))
        print(self.center_text(f"Generated on: {datetime.now().strftime('%B %d, %Y')}", width))
        print(self.center_text("", width))
        print(bottom_border)
        print("🌟" * 20 + "\n")
    
    def animated_hello_world(self):
        """Display animated Hello World message"""
        message = "Hello World!"
        print("\nAnimated Greeting:")
        
        # Letter by letter animation
        for i in range(len(message) + 1):
            sys.stdout.write(f"\r✨ {message[:i]}")
            sys.stdout.flush()
            time.sleep(0.2)
        
        print(" ✨")
        time.sleep(0.5)
        
        # Sparkle effect
        sparkles = ["✨", "🌟", "⭐", "💫", "🌠"]
        for _ in range(5):
            for sparkle in sparkles:
                sys.stdout.write(f"\r{sparkle} {message} {sparkle}")
                sys.stdout.flush()
                time.sleep(0.3)
        
        print(f"\n🎉 {message} 🎉\n")
    
    def interactive_menu(self):
        """Display interactive menu for different card options"""
        while True:
            print("\n" + "=" * 40)
            print("🎁 HELLO WORLD GREETING CARD MENU 🎁")
            print("=" * 40)
            print("1. Display Standard Greeting Card")
            print("2. Show Animated Hello World")
            print("3. Create Custom Message")
            print("4. Exit")
            print("=" * 40)
            
            try:
                choice = input("Select an option (1-4): ").strip()
                
                if choice == "1":
                    self.display_hello_world_card()
                elif choice == "2":
                    self.animated_hello_world()
                elif choice == "3":
                    self.create_custom_card()
                elif choice == "4":
                    print("\n👋 Thank you for using the Greeting Card Generator!")
                    print("🌟 Goodbye! 🌟\n")
                    break
                else:
                    print("❌ Invalid option. Please choose 1-4.")
                    
            except KeyboardInterrupt:
                print("\n\n👋 Goodbye!")
                break
            except Exception as e:
                print(f"❌ Error: {e}")
    
    def create_custom_card(self):
        """Allow user to create a custom greeting card"""
        print("\n🎨 Custom Greeting Card Creator")
        
        try:
            message = input("Enter your custom message: ").strip()
            if not message:
                message = "Hello World!"
            
            recipient = input("Enter recipient name (optional): ").strip()
            
            width = max(len(message) + 10, 40)
            top_border, bottom_border = self.create_border(width)
            
            print("\n🎁 Your Custom Greeting Card:")
            print(top_border)
            print(self.center_text("", width))
            print(self.center_text("🌟 CUSTOM GREETING CARD 🌟", width))
            print(self.center_text("", width))
            print(self.center_text(message, width))
            print(self.center_text("", width))
            
            if recipient:
                print(self.center_text(f"For: {recipient}", width))
                print(self.center_text("", width))
            
            print(self.center_text(f"Created: {datetime.now().strftime('%B %d, %Y at %I:%M %p')}", width))
            print(self.center_text("", width))
            print(bottom_border)
            print()
            
        except KeyboardInterrupt:
            print("\n❌ Card creation cancelled.")

def main():
    """Main function to run the greeting card generator"""
    card_generator = GreetingCard()
    
    print("🎉 Welcome to the Hello World Greeting Card Generator! 🎉")
    
    # Quick demonstration
    card_generator.display_hello_world_card()
    
    # Ask if user wants interactive mode
    try:
        response = input("Would you like to explore more options? (y/n): ").strip().lower()
        if response in ['y', 'yes']:
            card_generator.interactive_menu()
        else:
            print("\n🌟 Thank you for trying the Hello World Greeting Card! 🌟")
    except KeyboardInterrupt:
        print("\n\n👋 Goodbye!")

if __name__ == "__main__":
    main()

File: greeting_card.js
#!/usr/bin/env node
/**
 * Greeting Card Generator in JavaScript
 * A Node.js script to generate and display a greeting card message with "Hello There" text.
 */

const readline = require('readline');

// ANSI color codes for terminal styling
const colors = {
    reset: '\x1b[0m',
    bright: '\x1b[1m',
    dim: '\x1b[2m',
    red: '\x1b[31m',
    green: '\x1b[32m',
    yellow: '\x1b[33m',
    blue: '\x1b[34m',
    magenta: '\x1b[35m',
    cyan: '\x1b[36m',
    white: '\x1b[37m',
    bg_red: '\x1b[41m',
    bg_green: '\x1b[42m',
    bg_yellow: '\x1b[43m',
    bg_blue: '\x1b[44m',
    bg_magenta: '\x1b[45m',
    bg_cyan: '\x1b[46m'
};

/**
 * Clear the terminal screen
 */
function clearScreen() {
    console.clear();
}

/**
 * Print a decorative border
 * @param {number} width - Width of the border
 * @param {string} color - Color for the border
 */
function printBorder(width = 50, color = colors.cyan) {
    console.log(color + "+" + "=".repeat(width - 2) + "+" + colors.reset);
}

/**
 * Print text centered within the given width
 * @param {string} text - Text to center
 * @param {number} width - Width of the container
 * @param {string} color - Color for the text
 */
function printCenteredText(text, width = 50, color = colors.white) {
    const padding = Math.floor((width - text.length - 2) / 2);
    const rightPadding = width - text.length - padding - 2;
    console.log(colors.cyan + "|" + color + " ".repeat(padding) + text + " ".repeat(rightPadding) + colors.cyan + "|" + colors.reset);
}

/**
 * Print an empty line with borders
 * @param {number} width - Width of the line
 */
function printEmptyLine(width = 50) {
    console.log(colors.cyan + "|" + " ".repeat(width - 2) + "|" + colors.reset);
}

/**
 * Sleep for a given number of milliseconds
 * @param {number} ms - Milliseconds to sleep
 */
function sleep(ms) {
    return new Promise(resolve => setTimeout(resolve, ms));
}

/**
 * Animate text by printing character by character
 * @param {string} text - Text to animate
 * @param {number} delay - Delay between characters in ms
 */
async function animateText(text, delay = 100) {
    for (const char of text) {
        process.stdout.write(char);
        await sleep(delay);
    }
    console.log();
}

/**
 * Display the main greeting card
 */
function displayGreetingCard() {
    clearScreen();
    
    const cardWidth = 50;
    
    // Top border
    printBorder(cardWidth);
    printEmptyLine(cardWidth);
    
    // Decorative elements
    printCenteredText("🌟 ✨ 🌟 ✨ 🌟", cardWidth, colors.yellow);
    printEmptyLine(cardWidth);
    
    // Main message
    printCenteredText("HELLO THERE!", cardWidth, colors.bright + colors.magenta);
    printEmptyLine(cardWidth);
    
    // Sub message
    printCenteredText("Hope you have a wonderful day!", cardWidth, colors.green);
    printEmptyLine(cardWidth);
    
    // More decorative elements
    printCenteredText("🎉 🎈 🎉 🎈 🎉", cardWidth, colors.yellow);
    printEmptyLine(cardWidth);
    
    // Bottom border
    printBorder(cardWidth);
}

/**
 * Display an animated version of the greeting card
 */
async function animatedGreetingCard() {
    clearScreen();
    console.log("\n".repeat(5));
    
    // Animated greeting
    process.stdout.write(" ".repeat(15));
    await animateText(colors.bright + colors.yellow + "✨ Hello There! ✨" + colors.reset, 200);
    
    await sleep(1000);
    
    process.stdout.write(" ".repeat(10));
    await animateText(colors.bright + colors.green + "🌟 Hope you have a wonderful day! 🌟" + colors.reset, 100);
    
    console.log("\n".repeat(3));
}

/**
 * Generate a simple ASCII art greeting
 */
function displayAsciiGreeting() {
    clearScreen();
    console.log(colors.bright + colors.cyan + `
    ╔══════════════════════════════════════════════╗
    ║                                              ║
    ║    ██╗  ██╗███████╗██╗     ██╗      ██████╗   ║
    ║    ██║  ██║██╔════╝██║     ██║     ██╔═══██╗  ║
    ║    ███████║█████╗  ██║     ██║     ██║   ██║  ║
    ║    ██╔══██║██╔══╝  ██║     ██║     ██║   ██║  ║
    ║    ██║  ██║███████╗███████╗███████╗╚██████╔╝  ║
    ║    ╚═╝  ╚═╝╚══════╝╚══════╝╚══════╝ ╚═════╝   ║
    ║                                              ║
    ║               ████████╗██╗  ██╗███████╗██████╗ ███████╗   ║
    ║               ╚══██╔══╝██║  ██║██╔════╝██╔══██╗██╔════╝   ║
    ║                  ██║   ███████║█████╗  ██████╔╝█████╗     ║
    ║                  ██║   ██╔══██║██╔══╝  ██╔══██╗██╔══╝     ║
    ║                  ██║   ██║  ██║███████╗██║  ██║███████╗   ║
    ║                  ╚═╝   ╚═╝  ╚═╝╚══════╝╚═╝  ╚═╝╚══════╝   ║
    ║                                              ║
    ║              Hope you have a wonderful day!  ║
    ║                                              ║
    ╚══════════════════════════════════════════════╝
    ` + colors.reset);
}

/**
 * Main function to run the greeting card program
 */
async function main() {
    const rl = readline.createInterface({
        input: process.stdin,
        output: process.stdout
    });

    console.log(colors.bright + colors.blue + "Welcome to the Greeting Card Generator!" + colors.reset);
    console.log("Choose an option:");
    console.log("1. Static Greeting Card");
    console.log("2. Animated Greeting Card");
    console.log("3. ASCII Art Greeting");
    console.log("4. All variations");

    rl.question("\nEnter your choice (1, 2, 3, or 4): ", async (choice) => {
        try {
            switch (choice.trim()) {
                case "1":
                    displayGreetingCard();
                    break;
                case "2":
                    await animatedGreetingCard();
                    break;
                case "3":
                    displayAsciiGreeting();
                    break;
                case "4":
                    displayGreetingCard();
                    console.log("\nPress Enter to see the animated version...");
                    rl.question("", async () => {
                        await animatedGreetingCard();
                        console.log("Press Enter to see the ASCII art version...");
                        rl.question("", () => {
                            displayAsciiGreeting();
                            rl.close();
                        });
                    });
                    return;
                default:
                    console.log("Invalid choice. Displaying static greeting card...");
                    displayGreetingCard();
                    break;
            }
        } catch (error) {
            console.error(`An error occurred: ${error.message}`);
            console.log("Displaying default greeting card...");
            displayGreetingCard();
        }
        
        rl.close();
    });
}

// Run the program if this file is executed directly
if (require.main === module) {
    main().catch(console.error);
}

module.exports = {
    displayGreetingCard,
    animatedGreetingCard,
    displayAsciiGreeting
};